src/lib/controllers/ui-controller.js

```diff
@@ -1227,7 +1227,10 @@ export default class UIController {
     this.updatePageAnnotationData(data.annotations)
   }
 
-  onWordItemSelected (homonym) {
+  async onWordItemSelected (wordItem) {
+    let wordItemFull = await this.userDataManager.query({ dataType: 'WordItem', params: { wordItem } }, { type: 'full' })
+    console.info('****************onWordItemSelected', wordItemFull[0].homonym)
+    let homonym = wordItemFull[0].homonym
```

Comment:
so at this point, we could probably check to see if the homonym is fully populated and if so skip the lexical query ? (Not  needed to  change for this PR, but just making sure I understand what is happening here)

Reply:
Yes, we could do this check on double-click the word inside `newLexicalRequest` I believe - need to be checked.